src/Microsoft.AspNetCore.Mvc.Core/Internal/ControllerActionInvokerCache.cs

```diff
@@ -79,30 +83,42 @@ public IFilterMetadata[] GetFilters(ActionContext actionContext)
 
             if (allFiltersCached)
             {
-                entry = new CacheEntry(filters);
+                entry = new ControllerActionInvokerCacheEntry(filters,null,executor);
             }
             else
             {
-                entry = new CacheEntry(items);
+                entry = new ControllerActionInvokerCacheEntry(null,items,executor);
             }
 
             cache.Entries.TryAdd(actionDescriptor, entry);
-            return filters;
+            return entry;
         }
 
-        private IFilterMetadata[] GetFiltersFromEntry(CacheEntry entry, ActionContext actionContext)
+        public IFilterMetadata[] GetFilters(ControllerContext controllerContext)
         {
-            Debug.Assert(entry.Filters != null || entry.Items != null);
+            var entry = GetActionInvokerCachedItem(controllerContext);
+            return GetFiltersFromEntry(entry, controllerContext);
+        }
+
+        public ObjectMethodExecutor GetControllerActionMethodExecutor(ControllerContext controllerContext)
```

Comment:
Don't do this. Just change the `ControllerActionInvoker` to look up the entry once. The point of putting these both in the cache together is to avoid the double-lookup.